### PR TITLE
topsql: rename topsql related variable and remove useless variable

### LINKED ISSUE #31433: Publish or remove Top SQL variables
## Enhancement

According to an accepted RFC https://github.com/pingcap/tidb/blob/master/docs/design/2021-07-29-hidden-sysvars.md we should consider removing Top SQL related hidden variables. Previously these hidden variables are hidden mainly for their advanced usages. As suggested in the RFC, these hidden variables should be either carefully checked and then release to the public, or removed.

My suggestion is:

| Variable | Action |
|---------|-------|
| tidb_top_sql_precision_seconds | Remove from the variable, as changing it is very likely to cause problems |
| tidb_top_sql_max_statement_count | Rename to reduce ambiguity + Public: tidb_top_sql_max_time_series_count |
| tidb_top_sql_max_collect | Rename to reduce ambiguity + Public: tidb_top_sql_max_meta_count |
| tidb_top_sql_report_interval_seconds | Remove from the variable, as changing it is very likely to cause problems |

### PR DESCRIPTION
**Pull request by crazycs520**
Signed-off-by: crazycs520 <crazycs520@gmail.com>

### What problem does this PR solve?

Issue Number: close #31433

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

Documentation

- [x] Contains variable changes

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by crazycs520** [comment]
@breeswish PTAL

File: sessionctx/variable/removed.go [hunk 1]
```diff
@@ -22,10 +22,14 @@ package variable
...
-	TiDBSlowLogMasking:             "use tidb_redact_log instead",
-	PlacementChecks:                "placement_checks is removed and use tidb_placement_mode instead",
+	TiDBEnableAlterPlacement:        "alter placement is now always enabled",
+	TiDBEnableGlobalTemporaryTable:  "temporary table support is now always enabled",
+	TiDBSlowLogMasking:              "use tidb_redact_log instead",
+	PlacementChecks:                 "placement_checks is removed and use tidb_placement_mode instead",
+	TiDBTopSQLPrecisionSeconds:      "not support change it now",
+	TiDBTopSQLReportIntervalSeconds: "not support change it now",
+	TiDBTopSQLMaxStatementCount:     "use tidb_top_sql_max_time_series_count instead",
+	TiDBTopSQLMaxCollect:            "use tidb_top_sql_max_meta_count instead",
```
Comment (breeswish) [inline sessionctx/variable/removed.go:32]: These variables are never provided to users. Maybe there is no need to notify them at all.

File: sessionctx/variable/sysvar.go [hunk 2]
```diff
@@ -1241,17 +1241,7 @@ var defaultSysVars = []*SysVar{
...
-	}, SetGlobal: func(vars *SessionVars, s string) error {
-		val, err := strconv.ParseInt(s, 10, 64)
-		if err != nil {
-			return err
-		}
-		topsqlstate.GlobalState.PrecisionSeconds.Store(val)
-		return nil
-	}},
-	{Scope: ScopeGlobal, Name: TiDBTopSQLMaxStatementCount, Value: strconv.Itoa(topsqlstate.DefTiDBTopSQLMaxStatementCount), Type: TypeInt, Hidden: true, MinValue: 0, MaxValue: 5000, GetGlobal: func(s *SessionVars) (string, error) {
+	{Scope: ScopeGlobal, Name: TiDBTopSQLMaxTimeSeriesCount, Value: strconv.Itoa(topsqlstate.DefTiDBTopSQLMaxTimeSeriesCount), Type: TypeInt, Hidden: true, MinValue: 0, MaxValue: 5000, GetGlobal: func(s *SessionVars) (string, error) {
```
Comment (breeswish) [inline sessionctx/variable/sysvar.go:1244]: Is it correct for min value to be 0?

File: sessionctx/variable/sysvar.go [hunk 3]
```diff
@@ -1261,7 +1251,7 @@ var defaultSysVars = []*SysVar{
 		topsqlstate.GlobalState.MaxStatementCount.Store(val)
 		return nil
 	}},
-	{Scope: ScopeGlobal, Name: TiDBTopSQLMaxCollect, Value: strconv.Itoa(topsqlstate.DefTiDBTopSQLMaxCollect), Type: TypeInt, Hidden: true, MinValue: 1, MaxValue: 10000, GetGlobal: func(s *SessionVars) (string, error) {
+	{Scope: ScopeGlobal, Name: TiDBTopSQLMaxMetaCount, Value: strconv.Itoa(topsqlstate.DefTiDBTopSQLMaxMetaCount), Type: TypeInt, Hidden: true, MinValue: 1, MaxValue: 10000, GetGlobal: func(s *SessionVars) (string, error) {
```
Comment (breeswish) [inline sessionctx/variable/sysvar.go:1254]: Hidden should be false.

**Review by breeswish — COMMENTED** [review]
Please also add tests to verify that they are not hidden any more.

File: server/tidb_test.go [hunk 4]
```diff
@@ -130,9 +130,9 @@ func createTidbTestTopSQLSuite(t *testing.T) (*tidbTestTopSQLSuite, func()) {
 	}()
 
 	dbt := testkit.NewDBTestKit(t, db)
-	dbt.MustExec("set @@global.tidb_top_sql_precision_seconds=1;")
-	dbt.MustExec("set @@global.tidb_top_sql_report_interval_seconds=2;")
-	dbt.MustExec("set @@global.tidb_top_sql_max_statement_count=5;")
+	topsqlstate.GlobalState.PrecisionSeconds.Store(1)
+	topsqlstate.GlobalState.ReportIntervalSeconds.Store(2)
```
Comment (breeswish) [inline server/tidb_test.go:134]: These two tests permently modifies the topsqlstate.GlobalState. Will it affect and pollute other tests?

**Review by breeswish — APPROVED** [review]
The rest LGTM

File: server/tidb_test.go (hunk 4 above)
Comment (crazycs520) [reply server/tidb_test.go:134]: All topsql related tests in this pkg will use this to setup,  so it's ok here?

**Review by zhongzc — APPROVED** [review]

**Comment by crazycs520** [comment]
@morgo PTAL

**Review by morgo — APPROVED** [review]
I have the same question as @breeswish on the tidb_test.go file. But the sysvar changes LGTM.

File: server/tidb_test.go (hunk 4 above)
Comment (crazycs520) [reply server/tidb_test.go:134]: ok, already add restore default logic.

**Comment by crazycs520** [comment]
> I have the same question as @breeswish on the tidb_test.go file. But the sysvar changes LGTM.

ok, already resolved.

**Pull request merged**